microsetta_private_api/repo/qiita_repo.py

```diff
@@ -84,6 +112,9 @@ def push_metadata_to_qiita(self, barcodes=None):
         # calls to this function if and as needed.
         to_push = list(barcodes - samples_in_qiita)[:1000]
 
+        # lock survey-sample association
+        self.lock_completed_surveys_to_barcodes(barcodes)
```

Comment:
```suggestion
        self.lock_completed_surveys_to_barcodes(to_push)
```

Need to limit it to only the barcodes actually being pushed to Qiita.